docs/samples/s3/README.md

```diff
@@ -0,0 +1,117 @@
+## Setup
+1. Your ~/.kube/config should point to a cluster with [KFServing installed](https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md#deploy-kfserving).
+2. Your cluster's Istio Ingress gateway must be network accessible.
+3. Your cluster's Istio Egresss gateway must [allow accessing S3 Storage](https://knative.dev/docs/serving/outbound-network-access/)
+4. The example uses the Kubeflow's Minio setup if you have [Kubeflow](https://www.kubeflow.org/docs/started/getting-started/) installed,
+you can also setup your own [Minio server](https://docs.min.io/docs/deploy-minio-on-kubernetes.html) or use other S3 compatible cloud storage.
+
+
+## Train TF mnist model and save on S3
+Follow Kubeflow's [TF mnist example](https://github.com/kubeflow/examples/tree/master/mnist#using-s3) to train a TF mnist model and save on S3,
+change following S3 access settings, `modelDir` and `exportDir` as needed. If you already have a mnist model saved on S3 you can skip this step.
+```bash
+export S3_USE_HTTPS=0 #set to 0 for default minio installs
+export S3_ENDPOINT=minio-service.kubeflow:9000
+export AWS_ENDPOINT_URL=http://${S3_ENDPOINT}
+
+kustomize edit add configmap mnist-map-training --from-literal=S3_ENDPOINT=${S3_ENDPOINT}
+kustomize edit add configmap mnist-map-training --from-literal=AWS_ENDPOINT_URL=${AWS_ENDPOINT_URL}
+kustomize edit add configmap mnist-map-training --from-literal=S3_USE_HTTPS=${S3_USE_HTTPS}
+
+kustomize edit add configmap mnist-map-training --from-literal=modelDir=s3://mnist/v1
+kustomize edit add configmap mnist-map-training --from-literal=exportDir=s3://mnist/v1/export
+```
+
+## Create S3 Secret
+If you already have a S3 secret created you can skip this step, since `KFServing` is relying on secret annotations to setup proper
+S3 environment variables you may still need to add following annotations to your secret to overwrite S3 endpoint or other S3 options.
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+  annotations:
+     serving.kubeflow.org/s3-endpoint: minio-service.kubeflow:9000 # replace with your s3 endpoint
+     serving.kubeflow.org/s3-usehttps: "0" # by default 1, for testing with minio you need to set to 0
+type: Opaque
+data:
+  awsAccessKeyID: XXXX
+  awsSecretAccessKey: XXXXXXXX
+```
+Create or update the secret
+```bash
+kubectl apply -f s3_secret.yaml 
+```
+
+## Attach S3 Secret to Service Account
+`KFServing` gets the S3 secrets from service account, it requires user to append the above created or existing secret to your service account's secret list.
+By default `KFServing` uses **default** service account in the namespace where `KFService` deploys to, user can also specify the service account on `KFService` CRD.
+
+```bash
+kubectl edit sa default
+```
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: default # change to your service account name
+secrets:
+- name: default-token-zjpsv
+- name: mysecret # append the s3 secret here and save
+```
```

Comment:
Same here - this is offloading and exposing too much of kube backend to end users?

Reply:
Any thoughts how to improve the experience ? We need some way to create secret and attach the secret to service account.

Reply:
Ok - so I was discussing with @ellis-bigelow to move to "One" yaml semantic, and possibly align with MLSPec. Given that this is how currently things are, we shall get this PR in. I will bring it up on our next weekly sync what would be ideal experience there.